sandy/decay.py

```diff
@@ -148,6 +156,54 @@ def get_decayconstant(self, with_uncertainty=True):
         else:
             return df.LAMBDA
 
+    def custom_perturbation_decayconstant(self, pert):
+        """
+        Apply a custom perturbation to the decay constant.
+
+        Parameters
+        ----------
+        pert : `float`
+            Perturbation coefficient as ratio value.
+
+        Returns
+        -------
+        `sandy.decay.DecayData`
+            DacayData instance with given ZAM and decay constant perturbed.
+
+        Examples
+        --------
+        >>> endf6 = sandy.get_endf6_file("jeff_33", 'decay', 922350)
+        >>> rdd = sandy.DecayData.from_endf6(endf6)
+        >>> dc_pert = rdd.custom_perturbation_decayconstant(0.05)
+        >>> dc_pert.data
+        {922350: {'half_life': 2.22102e+16,
+                  'decay_constant': 3.2768932273817556e-17,
+                  'decay_constant_uncertainty': 2.2171470275223715e-20,
+                  'stable': False,
+                  'spin': 3.5,
+                  'parity': -1.0,
+                  'decay_energy': {'beta': 50671.7,
+                   'gamma': 163616.0,
+                   'alpha': 4464600.0,
+                   'total': 4678887.7},
+                  'decay_energy_uncertainties': {'beta': 4291.63,
+                   'gamma': 1708.01,
+                   'alpha': 163255.0,
+                   'total': 163320.33067324167},
+                  'decay_modes': {'40x0': {'decay_products': {902310: 1.0, 20040: 1.0},
+                    'branching_ratio': 1.0,
+                    'branching_ratio_uncertainty': 0.0001},
+                   '60x0': {'decay_products': {},
+                    'branching_ratio': 7.2e-11,
+                    'branching_ratio_uncertainty': 2.1e-11}}}}
+        """
+        dc = self.get_decayconstant().LAMBDA
+        dc = dc * (1 + pert)

```

Comment:
We are missing the ZAM here. As mentioned before we don't want to perturb all decay constants.

Reply:
@EnricaBelfiore , is it clear why we want to perturb only one ZAM?

Reply:
Not so clear I think. It cannot be 'all', but why not a list?

Reply:
Sorry, stupid question. I wrote a notebook and I get it.

Reply:
If it were a list of zam, then I would expect a list of perturbations (we don't want to apply the same perturbation to all decay constants, right?! Imagine when we do random sampling).
But then why not have the function apply the perturbation to only one decay constant and the loop around it (maybe with another function). This follows the philosophy of having functions performing one and only one task.